conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,431 @@
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a nil-type-descriptor which is a simple-type-descriptor.
+Labels: let-expr, module-class-defn, nil-literal, nil-type, return-stmt, union-type, boolean, record-type,
+        list-constructor-expr, mapping-constructor-expr, table-constructor-expr, query-expr
+        object-constructor-expr, new-expr, function-call-expr, method-call-expr, error-constructor-expr,
+        type-cast-expr, typeof-expr, unary-logical-expr, is-expr, equality-expr, logical-expr, conditional-expr
+
+class NilClass {
+    null n;
+    function init(null n) {
+        self.n = n;
+    }
+
+    function getNil() returns () {
+        return self.n;
+    }
+
+    function isNil(int? i) returns boolean {
+        return i == ();
+    }
+}
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    io:println(() == let null n = null in n); // @output true
+    io:println(() == let () n = () in n); // @output true
+    io:println(() == let null n = () in n); // @output true
+
+    io:println(let null n = () in [n, n, n]); // @output [null,null,null]
+    io:println(let null n = () in [n, [n, n]]); // @output [null,[null,null]]
+
+    io:println(let null nx = (), null ny = () in {"n": nx, n1: nx, ny, ["a"]: [nx, ny]});
+    // @output {"n":null,"n1":null,"ny":null,"a":[null,null]}
+    io:println(let null nx = (), null ny = () in {"n": {nx, ny}}); // @output {"n":{"nx":null,"ny":null}}
+    Record r = let null n = () in {i: n};
+    io:println(r); // @output {"i":null}
+
+    io:println(let null n = () in table key(n1) [{n1: "N1", n2: n}, {n1: "N2", n2: n}]);
+    // @output [{"n1":"N1","n2":null},{"n1":"N2","n2":null}]
+
+    //var obj = let null n = () in object {
```

Comment:
Any particular reason for commenting this out?

Reply:
There are several other places too. If they are bugs, shall we add them to a separate test case and mention the `Fail-Issue`. (ex:https://github.com/ballerina-platform/ballerina-spec/pull/950/files#diff-7edd5cfbc95a7326fc67d8779a489358ddba25c4b3de0c48469369cde6653059R59)

Reply:
Fixed